src/Microsoft.PowerShell.ConsoleHost/host/msh/NativeHost.cs

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Runtime.Loader;
+
+namespace Microsoft.PowerShell
+{
+    /// <summary>
+    /// Provides helper functions to faciliate calling managed code from a native PowerShell host.
+    /// </summary>
+    public static class NativeHost
```

Comment:
For this to be a public API, I think a more specific name (and one we're likely not to want to use for a different purpose) would be ideal.

But also it feels like this should probably be another method on either `UnmanagedPSEntry` or `ManagedPSEntry`, since those classes exist for this exact purpose

Reply:
do you want to rename "NativeHost" to "UnmanagedHost", "UnmanagedPSHost" or "UnmanagedPSHostHelper"? name your pick, I'll update the PR accordingly. If you feel like this should be moved to UnmanagedPSEntry I can put it in there, it just felt a bit out of place since there was pretty much a single entry point function in that class.

However, I'll go with whatever you think feels right, I don't really care how it is named and where it is placed in the end, as long as it is included somewhere :)

Reply:
Ideally I'd like to find consensus with @iSazonov and @daxian-dbw on what the best course of action is. I think my own preference is to put it on `UnmanagedPSEntry`, but want to discuss it briefly first just so we're comfortable with the API.

Also, I'm not sure what the overhead on testing this would be, but without one I imagine there's an increased likelihood of regression

Reply:
@rjmholt it may be a difficult one to test automatically, because it's meant to be called from an unmanaged host. However, we could make it callable from managed code, and maybe use a test similar to whatever is currently used to load assemblies in memory?

Reply:
> Ideally I'd like to find consensus with @iSazonov and @daxian-dbw on what the best course of action is. I think my own preference is to put it on `UnmanagedPSEntry`, but want to discuss it briefly first just so we're comfortable with the API.

Perhaps `UnmanagedPSEntry` is too specific that coming from Windows PowerShell. Do we really need always ConsoleHost dll in custom projects? If no it is better to place the new API in SMA.